core/modules/replica/Configuration.cc

```diff
@@ -23,818 +23,1732 @@
+            _tryParameter<size_t>(      "common", "request_buf_size_bytes") ||
+            _tryParameter<unsigned int>("common", "request_retry_interval_sec") ||
+            _tryParameter<size_t>(      "controller", "num_threads") ||
+            _tryParameter<unsigned int>("controller", "request_timeout_sec") ||
+            _tryParameter<unsigned int>("controller", "job_timeout_sec") ||
+            _tryParameter<unsigned int>("controller", "job_heartbeat_sec") ||
+            _tryParameter<size_t>(      "controller", "http_server_threads") ||
+            _tryParameter<uint16_t>(    "controller", "http_server_port") ||
+            _tryParameter<string>(      "controller", "empty_chunks_dir") ||
+            _tryParameter<size_t>(      "database", "services_pool_size") ||
+            _tryParameter<string>(      "database", "qserv_master_host") ||
+            _tryParameter<uint16_t>(    "database", "qserv_master_port") ||
+            _tryParameter<string>(      "database", "qserv_master_user") ||
+            _tryParameter<string>(      "database", "qserv_master_name") ||
+            _tryParameter<size_t>(      "database", "qserv_master_services_pool_size") ||
+            _tryParameter<string>(      "database", "qserv_master_tmp_dir") ||
+            _tryParameter<unsigned int>("xrootd", "auto_notify") ||
+            _tryParameter<string>(      "xrootd", "host") ||
+            _tryParameter<uint16_t>(    "xrootd", "port") ||
+            _tryParameter<unsigned int>("xrootd", "request_timeout_sec") ||
+            _tryParameter<string>(      "worker", "technology") ||
+            _tryParameter<size_t>(      "worker", "num_svc_processing_threads") ||
+            _tryParameter<size_t>(      "worker", "num_fs_processing_threads") ||
+            _tryParameter<size_t>(      "worker", "fs_buf_size_bytes") ||
+            _tryParameter<size_t>(      "worker", "num_loader_processing_threads") ||
+            _tryParameter<size_t>(      "worker", "num_exporter_processing_threads") ||
+            _tryParameter<size_t>(      "worker", "num_http_loader_processing_threads") ||
+            _tryParameter<uint16_t>(    "worker_defaults", "svc_port") ||
+            _tryParameter<uint16_t>(    "worker_defaults", "fs_port") ||
+            _tryParameter<string>(      "worker_defaults", "data_dir") ||
+            _tryParameter<uint16_t>(    "worker_defaults", "db_port") ||
+            _tryParameter<string>(      "worker_defaults", "db_user") ||
+            _tryParameter<uint16_t>(    "worker_defaults", "loader_port") ||
+            _tryParameter<string>(      "worker_defaults", "loader_tmp_dir") ||
+            _tryParameter<uint16_t>(    "worker_defaults", "exporter_port") ||
+            _tryParameter<string>(      "worker_defaults", "exporter_tmp_dir") ||
+            _tryParameter<uint16_t>(    "worker_defaults", "http_loader_port") ||
+            _tryParameter<string>(      "worker_defaults", "http_loader_tmp_dir");
```

Comment:
Efficiency is not a concern since this operation is done only once at a time of loading the configuration parameters from MySQL. The other concern is valid. I will consider refactoring this parser based on the transient schema which has type information for each parameter, and which servers as a primary source of parameters that are allowed to be found in the persistent (MySQL) state.